### PR TITLE
fix(telegram): wrap markdown tables in <pre> for monospace rendering

### PR DESCRIPTION
**Pull request by kaieriksen**
## Problem

After #67 (markdown → Telegram HTML), tables render as garbled proportional text — pipe characters don't align without monospace. Screenshots from Andrew show agents producing markdown tables that arrive looking broken.

## Fix

Detect pipe-delimited markdown table blocks (2+ lines where every line starts with `|` and the second is a separator row) and wrap them in `<pre>` tags. Telegram's `<pre>` is monospaced, so columns align.

Tables are captured as placeholders at step 3.5 — after code/link protection but before the blanket HTML-escape — so:
- `<pre>` tags survive the escape step
- Cell content is HTML-escaped
- Bold/italic inside cells stays literal (correct for `<pre>` — if you need formatted key-value pairs, use a list)

## What changed

- **`src/channels/telegramFormat.ts`**: new `wrapTables()` function + call at step 3.5
- **`tests/channels-telegramFormat.test.ts`**: +94 lines, 8 test cases

## Test plan

- [ ] `bun run typecheck` clean
- [ ] `bun test` passes (existing + new table tests)

### CODE REVIEW

File: tests/channels-telegramFormat.test.ts [hunk 1]
```diff
@@ -174,6 +174,100 @@ describe("markdownToTelegramHtml — block-level", () => {
...
+    expect(html).toContain("a &lt; b");
+    expect(html).toContain("a &amp; b");
+  });
+
+  test("single pipe in a line is NOT treated as a table", () => {
+    // A line with just `a | b` (no leading pipe) is prose, not a table row.
+    const md = "Voltage: 52 V | Current: 10 A";
+    const html = markdownToTelegramHtml(md);
+    // Should NOT be in <pre>.
+    expect(html).not.toContain("<pre>");
```
Comment (andrewagrahamhodges) [inline tests/channels-telegramFormat.test.ts:261]: 🔵 **Trivial**: Good test case for ignoring isolated pipes. This is exactly the kind of false positive that would be annoying if misidentified.

**Review by andrewagrahamhodges — APPROVED** [review]
## Code Review: fix(telegram): markdown tables

**Author:** @kaieriksen | **Files:** 2 | **Verdict:** Approve

This is a clean and effective fix for the garbled table rendering on Telegram. By wrapping detected table blocks in `<pre>` tags, we leverage Telegram's monospace font for proper alignment. The implementation avoids mangling other formatting by using the existing placeholder pattern.

### Findings

| # | Severity | File | Line | Finding |
|---|----------|------|------|---------|
| 1 | 🔵 Trivial | tests/channels-telegramFormat.test.ts | 177 | Robust test coverage for various table shapes and edge cases. |

### What's Good
- **Logic:** Targeting the separator row drastically reduces false positives from prose that happens to use pipe characters.
- **Security:** Content inside the `<pre>` tag is still HTML-escaped, maintaining protection against injection.
- **Consistency:** The solution fits perfectly into the existing placeholder-driven transformation pipeline.

<details>
<summary>📋 AI Fix Prompt (copy to your AI editor)</summary>

```
Review and approve PR #68 (fix(telegram): wrap markdown tables in <pre> for monospace rendering).

The changes are well-integrated and correctly handle table detection without breaking existing formatting.
```
</details>
